Sources/XCRemoteCache/Dependencies/DependencyProcessor.swift

```diff
@@ -102,6 +107,10 @@ class DependencyProcessorImpl: DependencyProcessor {
             return false
         }
 
+        if dependency.type == .derivedFile && dependency.url.lastPathComponent.contains("-Swift.h") {
```

Comment:
I think we should skip all `DERIVED_FILE_DIR` files. Are there any reasons why to include only `-Swift.h`? 
Note that for `DERIVED_FILE_DIR` build setting left intact, all files there would return `false` because a file is classified as `.intermediate` (see line 111/120).

```suggestion
        if dependency.type == .derivedFile {
```

Reply:
Nit: if you agree with the above comment, we could move `.derivedFile` to `irrelevantDependenciesType`.